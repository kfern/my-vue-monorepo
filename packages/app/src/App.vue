<template>
  <img alt="Vue logo" src="./assets/logo.png" />
  <HelloWorld msg="Welcome to Vue 3 + TypeScript + Danfo.js" />
  <HelloDanfo>
    <ul>
      <li>
        <a href="https://danfo.jsdata.org/" target="_BLANK"
          >Danfo.js Documentation</a
        >
      </li>
    </ul>
  </HelloDanfo>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import HelloDanfo from "./components/HelloDanfo.vue";
import { HelloWorld } from "@kfern/my_components";

export default defineComponent({
  name: "App",
  components: {
    HelloDanfo,
    HelloWorld,
  },
});
</script>

<style>
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 10px;
  width: 90%;
}
</style>
