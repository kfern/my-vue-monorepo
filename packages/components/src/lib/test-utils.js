const { Machine } = require("xstate");
const { createModel } = require("@xstate/test");

// @from: https://timdeschryver.dev/blog/generated-tests-with-xstate-and-cypress#configuring-the-states-for-cypress
const addTests = (machineDefinition, tests) => {
  return {
    ...machineDefinition,
    states: Object.entries(machineDefinition.states).reduce(
      (s, [stateKey, stateValue]) => {
        return {
          ...s,
          [stateKey]: {
            ...stateValue,
            meta: {
              ...stateValue.meta,
              test: tests[stateKey],
            },
          },
        };
      },
      {}
    ),
  };
};

export const getTestsModels = (
  machineDefinition,
  checkState,
  triggerEvents
) => {
  const testMachine = Machine(addTests(machineDefinition, checkState));
  const testModel = createModel(testMachine).withEvents(triggerEvents);
  return testModel;
};
