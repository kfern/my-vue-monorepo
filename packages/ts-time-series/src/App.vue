<template>
  <!-- <HelloWorld msg="Welcome to Vue 3 + TypeScript + Danfo.js" /> -->
  <img alt="Vue logo" src="./assets/logo.png" />
  <HelloDanfo>
    <ul>
      <li>
        <a href="https://danfo.jsdata.org/" target="_BLANK"
          >Danfo.js Documentation</a
        >
      </li>
    </ul>
  </HelloDanfo>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import HelloDanfo from "./components/HelloDanfo.vue";

// import { HelloWorld } from "@kfern/components";

export default defineComponent({
  name: "App",
  components: {
    HelloDanfo,
    // HelloWorld,
  },
});
</script>

<style>
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 10px;
  width: 90%;
}
</style>
