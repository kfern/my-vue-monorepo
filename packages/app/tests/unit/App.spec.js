import { shallowMount } from "@vue/test-utils";
import App from "@/App.vue";

describe("App.vue", () => {
  it("is stable", () => {
    const wrapper = shallowMount(App);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
