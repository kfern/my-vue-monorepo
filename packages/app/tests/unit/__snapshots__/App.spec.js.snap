// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue is stable 1`] = `
<img alt="Vue logo" src="">
<hello-world-stub msg="Welcome to Your Vue.js App"></hello-world-stub>
`;
